dotCMS/src/main/java/com/dotmarketing/util/VelocityUtil.java

```diff
@@ -287,151 +287,167 @@ public static ToolboxManager getToolboxManager () {
 
-		if (ADMIN_MODE) {
+	            // gets user id from request for mod user
+	            com.liferay.portal.model.User backendUser = null;
 
-			// Making sure you are viewing the latest list of languages on any
-			// admin mode
-			languages = APILocator.getLanguageAPI().getLanguages();
-			context.put("languages", languages);
+	            try {
+	                backendUser = com.liferay.portal.util.PortalUtil.getUser(request);
+	                // Skin skin = backendUser.getSkin();
+	                // context.put("USER_SKIN", skin.getSkinId());
+	                context.put("backendUser", backendUser);
+	            } catch (Exception nsue) {
+	                Logger.warn(VelocityUtil.class, "Exception trying yo getUser: " + nsue.getMessage(), nsue);
+	            }
 
-			// gets user id from request for mod user
-			com.liferay.portal.model.User backendUser = null;
+	            HttpSession session = request.getSession();
+	            context.put("directorURL", session.getAttribute(com.dotmarketing.util.WebKeys.DIRECTOR_URL));
+	            context.put("viewFoldersURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_FOLDERS_URL));
+	            context.put("previewPageURL", session.getAttribute(com.dotmarketing.util.WebKeys.PREVIEW_PAGE_URL));
+	            context.put("viewContentsURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_CONTENTS_URL));
+	            context.put("viewBrowserURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_BROWSER_URL));
 
-			try {
-				backendUser = com.liferay.portal.util.PortalUtil.getUser(request);
-				// Skin skin = backendUser.getSkin();
-				// context.put("USER_SKIN", skin.getSkinId());
-				context.put("backendUser", backendUser);
-			} catch (Exception nsue) {
-				Logger.warn(VelocityUtil.class, "Exception trying yo getUser: " + nsue.getMessage(), nsue);
-			}
 
-			HttpSession session = request.getSession();
-			context.put("directorURL", session.getAttribute(com.dotmarketing.util.WebKeys.DIRECTOR_URL));
-			context.put("viewFoldersURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_FOLDERS_URL));
-			context.put("previewPageURL", session.getAttribute(com.dotmarketing.util.WebKeys.PREVIEW_PAGE_URL));
-			context.put("viewContentsURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_CONTENTS_URL));
-			context.put("viewBrowserURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_BROWSER_URL));
 
+	        }
 
+	        context.put("host", host);
 
-		}
-
-		context.put("host", host);
+	   }
+	
+	   
+	public static void makeBackendContext(Context context, IHTMLPage htmlPage, String cmsTemplateInode, String idURI, HttpServletRequest request,
+			boolean ADMIN_MODE, boolean EDIT_MODE, boolean PREVIEW_MODE, Host host) throws DotDataException {
 
+	    
+	        PageMode mode = (EDIT_MODE) 
+	                ? PageMode.EDIT 
+	                : (PREVIEW_MODE) 
+	                    ? PageMode.PREVIEW 
+	                    : (ADMIN_MODE) 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)